src/lib/components/Swap/RoutingDiagram/index.tsx

```diff
@@ -0,0 +1,144 @@
+import { Plural, Trans } from '@lingui/macro'
+import { Currency, TradeType } from '@uniswap/sdk-core'
+import { FeeAmount } from '@uniswap/v3-sdk'
+import { ReactComponent as AutoRouterIcon } from 'assets/svg/auto_router.svg'
+import { ReactComponent as DotLine } from 'assets/svg/dot_line.svg'
+import { Badge, BadgeDark } from 'lib/components/Badge'
+import Column from 'lib/components/Column'
+import Row from 'lib/components/Row'
+import Rule from 'lib/components/Rule'
+import TokenImg from 'lib/components/TokenImg'
+import styled, { ThemedText } from 'lib/theme'
+import { useMemo } from 'react'
+import { InterfaceTrade } from 'state/routing/types'
+
+import { getTokenPath, routerGradientCss, RoutingDiagramEntry } from './utils'
+
+const Wrapper = styled(Column)`
+  padding: 0.25em;
+`
+
+const RouteRow = styled(Row)`
+  grid-template-columns: 1em 1.15em 1fr 1em;
+  min-width: 430px;
+`
+
+const RouteDetailsContainer = styled(Row)`
+  padding: 0.1em 0.5em;
+  position: relative;
+`
+
+const DottedLine = styled.div`
+  align-items: center;
+  display: flex;
+  opacity: 0.5;
+  position: absolute;
+  width: calc(100% - 1em);
+  z-index: 1;
+`
+
+const ShortDottedLine = styled(DottedLine)`
+  overflow: hidden;
+  position: relative;
+  width: 71px;
+`
+
+const DotColor = styled(DotLine)`
+  path {
+    stroke: ${({ theme }) => theme.secondary};
+  }
+`
+
+// Used to cutoff space between badges and dotted lines.
+const TransparentBadgeWrapper = styled(Badge)`
+  background-color: ${({ theme }) => theme.dialog};
+  padding: 0 4px;
+`
+
+const DetailsRow = styled(Row)`
+  display: grid;
+  grid-template-columns: 4.8125em 1fr;
+  width: 100%;
+`
+
+const StyledAutoRouterLabel = styled(ThemedText.ButtonSmall)`
+  color: #27ae60;
+  font-weight: 500;
+  line-height: 1rem;
+
+  ${routerGradientCss}
+`
+
+const StyledAutoRouterIcon = styled(AutoRouterIcon)`
+  height: 0.875em;
+  width: 0.875em;
+  :hover {
+    filter: brightness(1.3);
```

Comment:
Could this be an onHover color, instead of a custom css filter? What is this meant to be?

Reply:
i think we can just delete this because we're not using it as a link like we are in app?

Reply:
deleted